stringer.go

```diff
@@ -66,3 +54,34 @@ func (k key) String() string {
 	}
 	return k.t.String()
 }
+
+func (pl paramList) String() string {
+	args := make([]string, len(pl.Params))
+	for i, p := range pl.Params {
+		args[i] = p.String()
+	}
+	return fmt.Sprint(args)
```

Comment:
I'm restraining myself from golfing allocations here, but I want you to know that I know you're trolling me.